src/NzbDrone.Core/Localization/Core/en.json

```diff
@@ -0,0 +1,6 @@
+{
+  "language": "Language",
+  "uiLanguage": "UI Language",
+  "uiLanguageHelpText": "Language that Sonarr will use for UI",
+  "uiLanguageHelpTextWarning": "Browser Reload Required"
```

Comment:
@markus101 While I agree these would initially make for a better dev experience, and would improve code readability initially, I don't think it's sustainable.
If a string gets updated, does the key change too? This would mean updating it in every translation file.
I'm not sure exactly how weblate works, but this would normally mean the text comes up as a new string instead of just picking up changes.
If the key doesn't change, then you're left outdated text, which would just make for a more confusing dev experience.

Reply:
> If a string gets updated, does the key change too?

Yes.

> This would mean updating it in every translation file.

Yes, but on the flip side, if you use static keys and you change a translation from `{fileCount} files` to `{fileCount} episode files` every translation is now wrong and unless you audit every translation you simply don't know they're wrong. If you expect French and get English you're going to realize it's not translated, if you expect French and get a different French translation than an English user gets you won't know.

I know there is going to be some churn due to spelling, punctuation, etc, but overall it'll enforce other languages to follow English.

Reply:
Fair enough, I can do that.
Other issue is that currently all keys start with a lower case letter due to the way the API generates responses.
Are you ok with that staying the same? e.g. `"browser Reload Required": "Browser Reload Required"`
Or should the API be updated?

Reply:
We should be able to override that behaviour, so we can have keys in whatever case they're written in. I'll look at how we can achieve that.

Reply:
Thanks, that would be appreciated.
I have left the keys starting with lower case in the meantime.

Reply:
> > This would mean updating it in every translation file.
> 
> Yes, but on the flip side, if you use static keys and you change a translation from `{fileCount} files` to `{fileCount} episode files` every translation is now wrong and unless you audit every translation you simply don't know they're wrong. If you expect French and get English you're going to realize it's not translated, if you expect French and get a different French translation than an English user gets you won't know.

As an FYI, Weblate does handle this by marking changes to en as needing review in other languages (for non key changes). Weblate also auto updates the keys based on the master translation so you really only ever have to work in en translation. 

I agree with the use of keys over strings for ease and avoidance of constant double work when messing with strings, but weblate should be able to accommodate the other easily.

Reply:
This looks to be our easiest way to achieve it:

```diff
diff --git a/src/Sonarr.Api.V3/Localization/LocalizationResource.cs b/src/Sonarr.Api.V3/Localization/LocalizationResource.cs
index 0be5c7626..7ad219096 100644
--- a/src/Sonarr.Api.V3/Localization/LocalizationResource.cs
+++ b/src/Sonarr.Api.V3/Localization/LocalizationResource.cs
@@ -1,8 +1,41 @@
+using System;
 using System.Collections.Generic;
+using Newtonsoft.Json;
 using Sonarr.Http.REST;
 
 namespace Sonarr.Api.V3.Localization
 {
+    public class LocalizationResourceSerializer : JsonConverter
+    {
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            var resource = value as LocalizationResource;
+            writer.WriteStartObject();
+            writer.WritePropertyName("strings");
+            writer.WriteStartObject();
+
+            foreach (var r in resource.Strings)
+            {
+                writer.WritePropertyName(r.Key);
+                serializer.Serialize(writer, r.Value);
+            }
+
+            writer.WriteEndObject();
+            writer.WriteEndObject();
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override bool CanConvert(Type objectType)
+        {
+            return typeof(LocalizationResource).IsAssignableFrom(objectType);
+        }
+    }
+
+    [JsonConverter(typeof(LocalizationResourceSerializer))]
     public class LocalizationResource : RestResource
     {
         public Dictionary<string, string> Strings { get; set; }
```

Result (the last two added for testing):
```json
{
  "strings": {
    "language": "Language",
    "uiLanguage": "UI Language",
    "uiLanguageHelpText": "Language that Sonarr will use for UI",
    "uiLanguageHelpTextWarning": "Browser Reload Required",
    "Hello World": "Hello World",
    "hello {name}": "hello {name}"
  }
}
```

In the same file to make it easy to test, we could also flatten the object and get rid of the strings, but I'm not too worried about that.

Reply:
Added and flattened. 🙂